rfcs/text/0010-email.md

```diff
@@ -1,68 +1,44 @@
-| `email.attachments` | flattened | A flattened field for anything related to attachments. This allows objects being stored with all information for each file when you have multiple attachments |
-| `email.direction` | keyword | Direction of the message based on the sending and receving domains |
-| `email.sender.address` | wildcard | Senders email address |
-| `email.sender.domain` | wildcard | Domain of the sender |
-| `email.sender.top_level_domain` | keyword | Top level domain of the sender |
-| `email.sender.registered_domain` | wildcard | Registered domain of the sender |
-| `email.sender.subdomain` | keyword | Subdomain of the sender |
-| `email.message_id` | keyword | Internet message ID of the message |
-| `email.reply_to.address` | wildcard | Reply-to address |
-| `email.return_path.address` | wildcard | The return address for the message |
-| `email.size` | long | Total size of the message, in bytes, including attachments |
-| `email.subject` | wildcard | Subject of the message |
-| `email.recipients.addresses` | `keyword[]` | Recipient addresses |
-| `email.domains` | `keyword[]` | domains related to the email |
-
-
-Other ECS fields used together with email usecases:
-| field | description |
-| --- | --- |
-| `event.duration` | The duration related to the email event. Could be the total duration in Quarantine, how long the email took to send from source to destination etc |
-| `event.start` | When the email event started
-| `event.end` | When the email event ended
-| `process.name` | When the event is related to a server or client. Does not take MTA into account which is part of a ongoing discussion |
-| `network.protocol` | Type of email protocol used |
-| `tls.*` | Used for TLS related information for the connection to for example a SMTP server over TLS |
-
-
+| `email.from` | keyword (array) | Stores the `from` email address(es) |
+| `email.timestamp` | keyword (array) | The local date and time the message was written |
+| `email.to` | keyword (array) | The email address(es) of the message recipient(s) |
+| `email.subject` | keyword; `.text` text multi-field | A brief summary of the topic of the message |
+| `email.cc` | keyword (array) | The email address(es) of the carbon copy (CC) recipient(s) |
+| `email.bcc` | keyword (array) | The email address(es) of the blind carbon copy (CC) recipient(s) |
+| `email.content_type` | keyword | Information about how the message is to be displayed. Typically a MIME type |
+| `email.message_id` | wildcard | Unique identifier for the email message that refers to a particular version of a particular message |
+| `email.reply_to` | keyword | Address that replies should be delivered to |
+| `email.direction` | keyword | Direction of the message based on the sending and receiving domains |
+| `email.x_mailer` | keyword | What application was used to draft and send the original email.
+
+### Additional event categorization allowed values
```

Comment:
I agree we should include fields for emails attachments eventually. Still, I'm undecided whether to have them or not in this initial round of the `email.*` fields or add attachment fields in a separate proposal later.

If folks feel there's a strong case for including them from the start, I'm open to adding.

Part of my hesitation - I haven't thought of a great option for multiple attachments in a single document without resorting to an array of objects and using a `nested` field. We introduced `nested` recently in ECS for other use cases. However, I am still cautious about `nested` since a) users still misunderstand the behavior of nested fields, b) nested fields consume additional storage, and c) lack of nested agg support in Kibana for visualizations.

Reply:
Can we consider adding another ID field, for example` email.local-id`? Exchange uses it's own IDs in addition to message-id for example `internal-message-id` or `network-message-id`. Other email servers will have their own session IDs as well.

Reply:
That sounds like a good idea, @ericwbentley 

Could you use the "insert suggestion" option to share how you think that'd need to be structured?

Reply:
Without using language too specific towards one product or another I think this covers what the field would be used for

```suggestion
### Additional event categorization allowed values
| `email.local_id` | keyword | Stores the local ID of the email from the server that created the event and not persistent across server hops |
```

Reply:
@ericwbentley @peasead I had added `email.external_id`, but I like what you proposed with `email.local_id` better. 😄 

I adjust the description a bit: https://github.com/elastic/ecs/pull/1219/commits/6368af1cb3b1805d07fae332ad928b44fa2363a3. Let me know if you have any feedback.